Source/Scene/ImageryLayer.js

```diff
@@ -969,6 +1149,230 @@ define([
         }
     };
 
+    function MakeCommandOptions() {
+        this.context = undefined;
+        this.outputTexture = undefined;
+        this.imagery = undefined;
+        this.imageryLayer = undefined;
+        this.vertexArray = undefined;
+        this.projectedRectangle = undefined;
+        this.projectedTexture = undefined;
+        this.final = undefined;
+    }
+
+    var arbitraryReprojectAttributeIndices = {
+        position : 0,
+        projectedCoordinates : 1
+    };
+
+    var geographicCartographicScratch = new Cartographic();
+    var projectedScratch = new Cartesian3();
+
+    /**
+     * Enqueues commands re-projecting multiple textures to a {@link GeographicProjection} on the next update
+     * and generate mipmaps for the geographic texture.
+     */
+    ImageryLayer.prototype._multisourceReprojectTexture = function(frameState, imagery) {
+        var projectedTextures = imagery.projectedTextures;
+        var projectedRectangles = imagery.projectedRectangles;
+        var projectedTexturesLength = projectedTextures.length;
+        var someTexture = projectedTextures[0];
+
+        var width = someTexture.width;
+        var height = someTexture.height;
+
+        var outputTexture = imagery.texture = new Texture({
+            context : frameState.context,
+            width : width,
+            height : height,
+            pixelFormat : someTexture.pixelFormat,
+            pixelDatatype : someTexture.pixelDatatype,
+            preMultiplyAlpha : someTexture.preMultiplyAlpha
+        });
+
+        if (CesiumMath.isPowerOfTwo(width) && CesiumMath.isPowerOfTwo(height)) {
+            outputTexture.generateMipmap(MipmapHint.NICEST);
+        }
+
+        var projection = this._imageryProvider.tilingScheme.sourceProjection;
+        var context = frameState.context;
+
+        // Create vertex buffers
+        var positions = new Float32Array(ARBITRARY_PROJECTION_VERTICES_WIDTH * ARBITRARY_PROJECTION_VERTICES_HEIGHT * 2);
```

Comment:
Since the position and index buffers don't change for all projected imagery, can we cache them?

Reply:
You can add `positionsBuffer.vertexArrayDestroyable = false` to prevent the vertex array from releasing them.

Reply:
Same comment about the shader.

Reply:
The other reproject function could do the same.

Reply:
> can we cache them?

I'm guessing we would also want to manually reference count and destroy these then, I'll look at `TileTerrain.js`.

> Same comment about the shader.

Is this already handled by the shader cache?

Reply:
> Is this already handled by the shader cache?

Yes, but the shaders can be destroyed every 120 frames. If downloading the images is slow, it could be destroying and recreating the shader a bunch.

Reply:
> The other reproject function could do the same.

Aha, turns out the other reproject function already does this... I just didn't figure that out until now...